plugins/compiler/src/main/kotlin/PluginFactoryGenerator.kt

```diff
@@ -165,6 +193,65 @@ class PluginFactoryGenerator(private val codeGenerator: CodeGenerator) {
             add(")\n\n")
         }.build()
 
+    private fun getConfigArguments(pluginOptions: List<PluginOption>) =
+        pluginOptions.map { option ->
+            val isNullable = !option.isRequired && option.defaultValue == null
```

Comment:
Shouldn't the condition be `||`?

Reply:
I have reworked this by introducing a new `isNullable` property to `PluginOption` to capture the case where a property is nullable but has a different default value, in which case it should still be possible to deliberately set it to `null`.